packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page/definition-tab/property-list-card/shared/property-type-form/data-type-selector/custom-expected-value-builder/object-expected-value-builder.tsx

```diff
@@ -0,0 +1,308 @@
+import { usePropertyTypesContextValue } from "../../../../../../use-property-types-context-value";
+import { PropertyTypeFormValues } from "../../../property-type-form-values";
+import { DeleteExpectedValueModal } from "./array-expected-value-builder/delete-expected-value-modal";
+import { ExpectedValueBadge } from "./array-expected-value-builder/expected-value-badge";
+import { CustomExpectedValueSelector } from "./shared/custom-expected-value-selector";
+
+const StyledTableHeadCell = (props: TableCellProps) => (
+  <TableCell
+    {...props}
+    sx={{
+      py: 1,
+      px: 1.5,
+      lineHeight: "18px",
+      fontSize: 11,
+      fontWeight: 600,
+      borderBottom: ({ palette }) => `1px solid ${palette.gray[30]}`,
+      whiteSpace: "nowrap",
+    }}
+  />
+);
+
+const StyledTableBodyCell = (props: TableCellProps) => (
+  <TableCell
+    {...props}
+    sx={{
+      py: 1,
+      px: 1.5,
+      flexDirection: "row",
+      alignItems: "center",
+      borderBottom: ({ palette }) => `1px solid ${palette.gray[30]} !important`,
+    }}
+  />
+);
+
+export interface ObjectExpectedValueRowProps {
+  objectId: string;
+  propertyId: string;
+  allowArrays: boolean;
+  required: boolean;
+  propertyIndex: number;
+}
+
+export const ObjectExpectedValueRow: FunctionComponent<
+  ObjectExpectedValueRowProps
+> = ({ objectId, propertyId, allowArrays, required, propertyIndex }) => {
+  const [show, setShow] = useState(false);
+
+  const { types: propertyTypes } = usePropertyTypesContextValue();
+  const { setValue } = useFormContext<PropertyTypeFormValues>();
+
+  const property = propertyTypes?.[propertyId as VersionedUri];
+
+  useEffect(() => {
+    setShow(true);
+  }, []);
+
+  return (
+    <TableRow
+      sx={{
+        maxHeight: show ? 1 : 0,
+        transition: ({ transitions }) => transitions.create("max-height"),
+      }}
+    >
+      <StyledTableBodyCell>
+        <FontAwesomeIcon
+          icon={faAsterisk}
+          sx={(theme) => ({
+            color: theme.palette.gray[50],
+            verticalAlign: "middle",
+          })}
+        />
+        <Typography
+          variant="smallTextLabels"
+          component="span"
+          ml={1.5}
+          color={(theme) => theme.palette.gray[80]}
+        >
+          {property?.title}
+        </Typography>
+        <Chip
+          color="purple"
+          label="PROPERTY TYPE"
+          sx={{
+            ml: 1.5,
+            borderColor: "transparent",
+            [`.${chipClasses.label}`]: {
+              fontWeight: 600,
+              fontSize: 11,
+            },
+          }}
+          size="xs"
+        />
+      </StyledTableBodyCell>
+      <StyledTableBodyCell align="center">
+        <Checkbox
+          checked={allowArrays}
+          onChange={(evt) => {
+            setValue(
+              `flattenedCustomExpectedValueList.${objectId}.data.properties.${propertyIndex}.allowArrays`,
+              evt.target.checked,
+            );
+          }}
```

Comment:
This is kinda confused – the value is being passed in by a parent, but you're directly updating without going through the parent. I think we kinda want to pick one.